pkg/extprom/http/instrument_server.go

```diff
@@ -37,97 +36,63 @@ func NewNopInstrumentationMiddleware() InstrumentationMiddleware {
 }
 
 type defaultInstrumentationMiddleware struct {
-	requestDuration *prometheus.HistogramVec
-	requestSize     *prometheus.SummaryVec
-	requestsTotal   *prometheus.CounterVec
-	responseSize    *prometheus.SummaryVec
+	metrics *defaultMetrics
 }
 
 // NewInstrumentationMiddleware provides default InstrumentationMiddleware.
 // Passing nil as buckets uses the default buckets.
 func NewInstrumentationMiddleware(reg prometheus.Registerer, buckets []float64) InstrumentationMiddleware {
-	if buckets == nil {
-		buckets = []float64{0.001, 0.01, 0.1, 0.3, 0.6, 1, 3, 6, 9, 20, 30, 60, 90, 120, 240, 360, 720}
+	return &defaultInstrumentationMiddleware{
+		metrics: newDefaultMetrics(reg, buckets, []string{}),
 	}
-
-	ins := defaultInstrumentationMiddleware{
-		requestDuration: promauto.With(reg).NewHistogramVec(
-			prometheus.HistogramOpts{
-				Name:    "http_request_duration_seconds",
-				Help:    "Tracks the latencies for HTTP requests.",
-				Buckets: buckets,
-			},
-			[]string{"code", "handler", "method"},
-		),
-
-		requestSize: promauto.With(reg).NewSummaryVec(
-			prometheus.SummaryOpts{
-				Name: "http_request_size_bytes",
-				Help: "Tracks the size of HTTP requests.",
-			},
-			[]string{"code", "handler", "method"},
-		),
-
-		requestsTotal: promauto.With(reg).NewCounterVec(
-			prometheus.CounterOpts{
-				Name: "http_requests_total",
-				Help: "Tracks the number of HTTP requests.",
-			}, []string{"code", "handler", "method"},
-		),
-
-		responseSize: promauto.With(reg).NewSummaryVec(
-			prometheus.SummaryOpts{
-				Name: "http_response_size_bytes",
-				Help: "Tracks the size of HTTP responses.",
-			},
-			[]string{"code", "handler", "method"},
-		),
-	}
-	return &ins
 }
 
 // NewHandler wraps the given HTTP handler for instrumentation. It
-// registers four metric collectors (if not already done) and reports HTTP
+// registers five metric collectors (if not already done) and reports HTTP
```

Comment:
What's the reason behind in-flight? It's usually misleading metrics given 15s interval and requests finishing sooner.. but maybe we can try having it to double check

Reply:
@bwplotka my attention adding the in-flight is to allow users of Thanos Receive to get better insights in situations of high load, where the number of concurrent requests will be relevant. Also, later on there will be a "metered gate" to limit the amount of in-flight requests.

Reply:
This is actually quite useful to know given atm there is no notion of enqueued requests in receiver. This is particularly useful for troubleshooting issues with load/receiver getting requests that it is not completing.

Reply:
Split PR with in-flight requests metrics: #5424.